docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+            : parent is FileScopedNamespaceDeclarationSyntax fileScopedNamespaceDeclarationSyntax
+            ? fileScopedNamespaceDeclarationSyntax.Name.ToString()
+            : AutoImplementAttributeNameSpace;
+    }
+
+    private static InterfaceModel[] GetInterfaceModels(Compilation compilation, ClassDeclarationSyntax classDeclarationSyntax)
+    {
+        //Get class usings in order to build full interface name if needed
+        string[] usings = GetUsings(classDeclarationSyntax.SyntaxTree.GetRoot());
+        string[] interfacesNames = GetInterfacesNames(classDeclarationSyntax.AttributeLists);
+
+        List<InterfaceModel> ret = [];
+        foreach (string interfaceName in interfacesNames)
+        {
+            INamedTypeSymbol? interfaceSymbol = GetInterfaceSymbol(compilation, usings, interfaceName);
+            if (interfaceSymbol is null)
+                continue;
+
+            //Get interface usings, in order to be sure to have the needed usings for the properties that will be generated
+            string[] interfaceUsings = interfaceSymbol
+                .DeclaringSyntaxReferences
+                .FirstOrDefault()?
+                .GetSyntax()
+                .SyntaxTree
+                .GetRoot()
+                .DescendantNodes()
+                .OfType<UsingDirectiveSyntax>()
+                .Select(x => x.ToString())
+                .ToArray() ?? [];
+
+            IPropertySymbol[] interfaceProperties = interfaceSymbol
+                .GetMembers()
+                .OfType<IPropertySymbol>()
+                .ToArray();
+
+            InterfacePropertyModel[] propertyModels = interfaceProperties
+                .Select(interfaceProperty =>
+                {
+                    /*Using "interfaceProperty.Type" instead of "interfaceProperty.Type.Name" in order to not have error in specific cases.
+                    Es. the type "int" has Name "Int32", but writing "Int32" casue compilation error if we are not using "System" namespace*/
+                    string type = interfaceProperty.Type.ToString();
+                    return new InterfacePropertyModel(type, interfaceProperty.Name, interfaceProperty.SetMethod is not null);
+                })
+                .ToArray();
+
+            string containingNamespace = interfaceSymbol.ContainingNamespace.ToString();
+
+            ret.Add(new InterfaceModel(interfaceName, containingNamespace, interfaceUsings, propertyModels));
+        }
+
+        return [.. ret];
+    }
+    private static string[] GetUsings(SyntaxNode root)
+    {
+        return root.DescendantNodes()
+                .OfType<UsingDirectiveSyntax>()
+                .Select(usingDirective =>
+                {
+                    //get string "System" from string "using System;"
+                    string usingName = usingDirective.ToString().Split(' ').Last().TrimEnd(';');
+                    return usingName;
+                })
+                .ToArray();
+    }
```

Comment:
I would not try to copy usings over. Instead, fully qualify everything. There's no telling what will conflict in the rest of the user's file with your own code.